### PR TITLE
Migration of known user-agents

### PR DESCRIPTION
**Pull request by commuterjoy**
The previous (failed) strategy migrated all user-agents, bots and all, whereas this one will hopefully just catch mostly web browsers used by humans.

/cc @lc512k @matthew-andrews

### CODE REVIEW

File: server/controllers/forced-opt-in.js [hunk 1]
```diff
@@ -44,13 +44,13 @@ module.exports = function(req, res, next) {
 	try {
 		res.set('Cache-Control', config.NO_CACHE);
 		const locationWithParams = req.originalUrl.replace('/__anon-opt-in', '');
-		const referrer = req.get('Referer') || '';
-		const whitelist = /(drudgereport)/i;
+		const ua = req.get('User-Agent') || '';
+		const whitelist = /(Chrome|Firefox|Safari|MSIE|Edge|Opera|UCBrowser)/i;
```
Comment (matthew-andrews) [inline server/controllers/forced-opt-in.js]: GoogleBot sometimes has Safari in the UA :-(
https://support.google.com/webmasters/answer/1061943?hl=en

I think we should have a blacklist for known bots (perhaps just SocialFlow initially!) in addition to the whitelist for popular browsers and in the long term aim to eliminate the whitelist :-)

File: test/smoke.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ module.exports = [
 		}
 	}, {
 		headers: {
-			Referer: 'http://www.google.com/'
+			'User-Agent': 'GoogleBot 2.1',
```
Comment (matthew-andrews) [inline test/smoke.js]: can we have a test for SocialFlow specifically :see_no_evil:

File: server/controllers/forced-opt-in.js (hunk 1 above)
Comment (commuterjoy) [inline server/controllers/forced-opt-in.js]: ok, I've added a blacklist for the Googlebot.

> SocialFlow

I've added two unit tests to cover the two SocialFlow user-agents they told us about.

File: server/controllers/forced-opt-in.js [hunk 3]
```diff
@@ -44,13 +44,14 @@ module.exports = function(req, res, next) {
...
+		const blacklist = /(Googlebot)/i;
```
Comment (matthew-andrews) [inline server/controllers/forced-opt-in.js]: sorry, just to double protect us in the future … can this be `/(Googlebot|SocialFlow)/i` ?

File: test/smoke.js (hunk 2 above)
Comment (commuterjoy) [inline test/smoke.js]: ok! done!

**Comment by matthew-andrews** [comment]
+1 :ship:

**Pull request merged**